packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -14,7 +14,52 @@ import 'material.dart';
 
 const Duration _kBottomSheetDuration = const Duration(milliseconds: 200);
 const double _kMinFlingVelocity = 700.0;
-const double _kFlingVelocityScale = 1.0 / 300.0;
+const double _kCloseProgressThreshold = 0.5;
+
+class _BottomSheetDragController extends StatelessComponent {
+  _BottomSheetDragController({
+    Key key,
+    this.performance,
+    this.child,
+    this.childHeight
+  }) : super(key: key);
+
+  final Performance performance;
+  final Widget child;
+  final double childHeight;
+
+  bool get _dismissUnderway => performance.direction == AnimationDirection.reverse;
+
+  void _handleDragUpdate(double delta) {
+    if (_dismissUnderway)
+      return;
+    performance.progress -= delta / (childHeight ?? delta);
+  }
+
+  void _handleDragEnd(Offset velocity, BuildContext context) {
+    if (_dismissUnderway)
+      return;
+    if (velocity.dy > _kMinFlingVelocity)
+      performance.fling(velocity: -velocity.dy / childHeight).then((dynamic value) {
+        Navigator.of(context).pop(value);
```

Comment:
It looks like I do need to call Navigator.pop(), since TransitionRoute.didPop() doesn't call the subcalsses didPop() method on dismiss.

You're right that there's no value to return here.